tests/rptest/tests/under_replicated_metric_test.py

```diff
@@ -0,0 +1,151 @@
+        self.await_startup()
+
+        metric_checks = self.create_metric_checks()
+
+        self.run_validation(min_records=20000,
+                            enable_idempotence=False,
+                            producer_timeout_sec=60,
+                            consumer_timeout_sec=180)
+
+        # all partitions should be fully replicated
+        assert self.check_no_under_replicated(metric_checks)
+
+    @cluster(num_nodes=4)
+    @parametrize(acks=1)
+    @parametrize(acks=-1)
+    def test_under_replicated_metric_with_different_acks(self, acks):
+        self.start_redpanda(num_nodes=3)
+        # create topics
+        topics = []
+        for _ in range(0, 6):
+            topics.append(TopicSpec(partition_count=random.randint(1, 10)))
+        # chose one topic to run the main workload
+        DefaultClient(self.redpanda).create_topic(topics)
+        self.topic = random.choice(topics).name
+
+        rpk_producer = RpkProducer(self.test_context,
+                                   self.redpanda,
+                                   self.topic,
+                                   16384,
+                                   20000,
+                                   acks=acks)
+
+        metric_checks = self.create_metric_checks()
+        rpk_producer.start()
+        rpk_producer.wait()
+        rpk_producer.stop()
+
+        assert self.check_no_under_replicated(metric_checks)
+
+    @cluster(num_nodes=4)
+    @parametrize(acks=1)
+    @parametrize(acks=-1)
+    def test_under_replicated_when_node_is_down(self, acks):
+        # setup recovery read size to speed things up
+        self.start_redpanda(
+            num_nodes=3,
+            extra_rp_conf={"raft_recovery_default_read_size": 4194304})
+        topics = []
+        for _ in range(0, 6):
+            topics.append(TopicSpec(partition_count=random.randint(1, 10)))
+        # chose one topic to run the main workload
+        DefaultClient(self.redpanda).create_topic(topics)
+        self.topic = random.choice(topics).name
+
+        rpk_producer = RpkProducer(self.test_context,
+                                   self.redpanda,
+                                   self.topic,
+                                   128,
+                                   sys.maxsize,
+                                   acks=acks)
+
+        metric_checks = self.create_metric_checks()
+        rpk_producer.start()
+
+        node = random.choice(self.redpanda.nodes)
+        self.redpanda.stop_node(node)
+
+        def has_under_replicated_replicas():
+            return all(
+                c.evaluate([(
+                    "vectorized_cluster_partition_under_replicated_replicas",
+                    lambda _, new_value: int(new_value) > 0)])
+                for c in metric_checks)
+
+        def has_no_under_replicated_replicas():
+            return self.check_no_under_replicated(metric_checks)
+
+        wait_until(lambda: has_under_replicated_replicas, 30, 1,
```

Comment:
this `wait_until` (and the one below, too) will always return success immediately because the lambda function evaluates to a function, not the result of the function:

```python
def f(): pass
x = lambda: f
assert x()
```

either pass in the function to be called:

```python
wait_until(has_under_replicated_replicas, 30, 1)
```

or have the lambda invoke the function:

```python
wait_until(lambda: has_under_replicated_replicas(), 30, 1)
```